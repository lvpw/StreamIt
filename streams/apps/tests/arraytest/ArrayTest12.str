/**
 * This tests at the intersection of arrays and global prop into
 * splitjoin weights.
 **/

static {
    int[2] A;

    init {
	A[0] = 2;
	A[1] = 2;
    }
}

void->void pipeline ArrayTest12 {
    add Source();
    add splitjoin {
	split roundrobin(A[1]);
	add Test(1);
	add Test(1);
	join roundrobin;
    }
    add Printer();
}

int->int filter Test(int j) {
    work pop 1 push 1 {
	push (j * pop());
    }
}

void->int filter Source {
    int i = 0;
    work push 1 { push(i++); }
}

int->void filter Printer {
    work pop 1 { println(pop()); }
}

