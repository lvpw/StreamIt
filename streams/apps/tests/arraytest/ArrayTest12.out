0
2
1
3
4
6
5
7
8
10
9
11
12
14
13
15
16
18
17
19
20
22
21
23
24
26
25
27
28
30
29
31
32
34
33
35
36
38
37
39
